lightning/src/ln/channelmanager.rs

```diff
@@ -3376,7 +3378,8 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 												incoming_packet_shared_secret: $htlc.prev_hop.incoming_packet_shared_secret,
 												phantom_shared_secret,
 											}), payment_hash,
-											HTLCFailReason::Reason { failure_code: 0x4000 | 15, data: htlc_msat_height_data }
+											HTLCFailReason::Reason { failure_code: 0x4000 | 15, data: htlc_msat_height_data },
+											None
```

Comment:
Hmm, we maybe need to rename the event for this case - we don't know the next hop here so it could just as well be a `PaymentReceiveFailed` (goes double for the similar line two hunks down). Maybe instead of `sink_node_id` we should have a `HTLCDestination` enum that has states like `OpenChannel` (with the node id and channel id), `Unknown` (with just the SCID we have), `Payment` (or something else that describes an HTLC to us), etc.

Reply:
makes sense. I'll look into this.